src/vm/contexts.rs

```diff
@@ -671,10 +664,20 @@ impl <'a,'b> Environment <'a,'b> {
 
         if make_read_only {
             self.global_context.roll_back();
-            result
-        } else {
-            self.global_context.handle_tx_result(result)
+            return result
         }
+
+        let result = self.global_context.handle_tx_result(result);
```

Comment:
I think this addresses your earlier comment about the special-cases code path not working within nested `contract-call?`'s?  If so, can you add a test case to the PoX tests that tests nested invocations of the PoX lock-up contract?

Reply:
I haven't seen a test for this yet.  Is there one?

Reply:
Test case added with https://github.com/blockstack/stacks-blockchain/pull/1850/commits/4374a9cbd9fd4fd3dd992e25b1bb57369ffe0de4, and I actually reverted my change, it turns out your initial approach was correct as well.